porch/func/evaluator/evaluator.proto

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package evaluator;
+
+import "struct.proto";
+
+option go_package = "github.com/GoogleContainerTools/kpt/porch/func/evaluator";
+
+// Evaluator of kpt functions
+service FunctionEvaluator {
+  // Evaluates a kpt function on the provided package
+  rpc EvaluateFunction(EvaluateFunctionRequest)
+      returns (EvaluateFunctionResponse) {}
+}
+
+message EvaluateFunctionRequest {
+  // Serialized ResourceList (https://kpt.dev/reference/schema/resource-list/)
+  bytes resource_list = 1;
+
+  // kpt image identifying the function to evaluate
+  string image = 2;
+
+  oneof config {
```

Comment:
IIRC in the `ResourceList.config.kubernetes.io/v1` - the `functionConfig` in `results` out are already part of the resource_list?

Wondering because, if the existing code on either end of the conversation will already have that marshaling, if it stays in the `bytes resource_list` you wouldn't need to have code to map between the proto and ResourceList types.

Reply:
You are right! I got carried away. This will simplify it considerably. Thank you!

Reply:
Fixed. Thank you! ptal